source/common/quic/http_datagram_handler.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include "envoy/http/codec.h"
+
+#include "source/common/common/assert.h"
+#include "source/common/common/logger.h"
+#include "source/common/http/header_map_impl.h"
+
+#include "absl/strings/string_view.h"
+#include "quiche/common/simple_buffer_allocator.h"
+#include "quiche/quic/core/http/quic_spdy_stream.h"
+
+namespace Envoy {
+namespace Quic {
+
+// Enables HTTP Datagrams and the Capsule Protocol support based on RFC 9297. This class is used
+// both on the receive side (HTTP/3 codec -> Envoy) and the send side (Envoy -> HTTP/3 codec) of
+// the QUIC client and server streams in Envoy.
+class HttpDatagramHandler : public quic::QuicSpdyStream::Http3DatagramVisitor,
+                            public quiche::CapsuleParser::Visitor,
+                            protected Logger::Loggable<Logger::Id::quic_stream> {
+public:
+  // Does not take ownership of the QuicSpdyStream. "stream" must refer to a valid object
+  // that outlives HttpDatagramHandler.
+  explicit HttpDatagramHandler(quic::QuicSpdyStream* stream);
+  ~HttpDatagramHandler() override;
+
+  // quic::QuicSpdyStream::Http3DatagramVisitor
+  // Normalizes HTTP/3 Datagrams to Capsules for Envoy to do subsequent processing.
```

Comment:
I think I would remove this comment (on line 29, not on line 28) and also the comment on line 35. Since these method are overrides, we can go look at the function definition to see how they're used. If we want to comment on the specific behavior of the override, that would be fine, though typically that would be a bit more specific than we have here. So I'd probably just drop it. But if you want to be a bit more specific, that'd be fine too.

Reply:
I agree. Done.